### ISSUE TITLE
Brainstorm contributor badges

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/team-compass#120]
At a recent team meeting @leportella had a neat idea that we could signal our "thanks" to others in the community in the form of "badges". Basically, a way to signal "thanks to XXX person for helping out in this way". We later had another conversation about how to in general say "thanks" to people for helping out in the JupyterHub repositories.

Are folks interested in running with this idea a bit, and brainstorm a system we can try to adopt? I think it sounds like a great idea.

(cc @KirstieJane as she also seemed interested in this, but I don't believe she gets notifications in this repo)

**Comment by leportella** [comment]
What came into my mind was something like this:

**Puppy badge** - reported an issue or opened a small pull request (or something like this)
**Mentor badge** - helped solving an issue or helping introducing new people to contributing
**Writer badge** - helped with documentation
**Awesome badge** - has been helping for a long time in a lot of different areas

Of course, we could have several types of badges if we like and names are completely up for suggestion. The idea would be something that people can earn one time or even evolve through.

**Comment by choldgraf** [comment]
nice 👍 

some others I thought of:

**First PR badge** - First time somebody merged a Pull Request
**Meetup badge** - Attend three monthly team meetings

Also what do you think about splitting Mentor badge into two:
**Community help badge** - the "helping solve an issue" part of your "mentor" badge above (and then mentor badge 
**Community growth badge** - helping onboard a new member to the community
**Mentor badge** - Serving as a mentor for a specific person/project (e.g. with Outreachy!)

Also, I wonder where this information would go? We could make a `badges.rst` file similar to the [teams page](https://jupyterhub-team-compass.readthedocs.io/en/latest/team.html). Could then be broken down by badge types and a list of people underneath each (it seems that might be better than listing people w/ all their badges underneath each person, since that might give the wrong idea that it's a competition or something)

what do you think?

**Comment by betatim** [comment]
What if the badge gets posted into the thread where you achieve it? We make a `badges.rst` with templates for all the badges and then anyone who realises you just achieved the XYZ badge can post it there.

The hard/time consuming/requires some spreadsheet to track things of the badges would be keeping track of people progress. To help with that: can we design badges that can be achieved within a single issue/discourse thread/PR?

**Comment by choldgraf** [comment]
Something like that sounds like a good idea - I do think a trick would be not adding so much labor to share / give badges that folks don't do it. I posted a few links that I found to help think through this.

What do you imagine as the structure of badges.rst? Would it be both a list of badge templates (e.g. copy/pasteable markdown snippets) as well as the list of recipients themselves?

**Comment by betatim** [comment]
The `badges.rst` would be the source of the templates. I hadn't thought of displaying them somewhere beyond where they were awarded. My feeling is that if you wanted to collect them somewhere we'd need a UI that is very slick for the awarder. I'm in GitHub and realise "award badge X -> open new tab -> surf to badge webpage -> authenticate myself -> enter github username -> select badge -> click submit" that is a lot of clicks. An alternative would be having badge templates as canned replies in GitHub so all I have to do is select it from a drop down. (This is a very GH focussed perspective on badges)

**Comment by choldgraf** [comment]
Something I just realized: Discourse has badges, and also lets you define custom badges that you can give out!

https://discourse.jupyter.org/badges

I wonder if we could make this platform our "badge system"

**Comment by leportella** [comment]
It could help indeed, but someone that is active here and not in discourse may suffer. But... I think we will always have ups and downs for each thing we choose :)

**Comment by choldgraf** [comment]
@leportella yeah that's a good point - though my hope would be that Discourse is a *more* inclusive place than GitHub, given that it's not code-driven, and given that you can sign up for Discourse without needing a GitHub account. What do you think?
